WQSearch.py

```diff
@@ -69,89 +81,170 @@ def _getZoneName(zone_id):
-    item_end = wqs_data.index(';', item_start)
-    item_json = wqs_data[item_start:item_end]
-
-    wqlist_item = json.loads(item_json)
-
-    for item in items_to_check:
-        if item.lower() in wqlist_item['name_enus'].lower():
-            yield (wqlist_item['name_enus'], item)
+# Wowhead returns the slot for the reward item as an integer ID.
+# This dict maps those IDs to human readable names, and allows for some fuzzy searching
+slot_ids = {
+    1: ["Head", "helm"],
+    3: ["Shoulder"],
+    5: ["Chest"],
+    6: ["Waist", "belt"],
+    7: ["Leg", "pants"],
+    8: ["Feet", "shoe", "boot"],
+    9: ["Wrist", "bracer"],
+    10: ["Hand", "glove"],
+    11: ["Finger", "ring"],
+    12: ["Trinket"],
+    13: ["One-hand", r"one.?hand", r"main.?hand", "weapon"],
+    14: ["Shield", r"off.?hand"],
+    15: ["Ranged", "gun", "bow", "weapon"],
+    16: ["Back", "cloak"],
+    17: ["Two-hand", r"two.?hand", "weapon"],
+    20: ["Trinket"],
+    23: ["Off-hand", r"off.?hand"],
+}
+
+
+def _checkForItems(item_id_list, wqs_data, items_to_check, slots_to_check):
+    for item_id in item_id_list:
+        # Takes an item id and looks up item in wqs_data.
+        patt = "_[" + str(item_id) + "]="
+        item_start = wqs_data.index(patt) + len(patt)
+        item_end = wqs_data.index(";", item_start)
+        item_json = wqs_data[item_start:item_end]
+
+        wqlist_item = json.loads(item_json)
+
+        # Get the slot ID of the item returned by wowhead
+        s_id = wqlist_item["jsonequip"].get("slotbak")
+        slot = ""
+        if s_id:
+            slot_names = slot_ids[s_id]
+            for s in slots_to_check:
+                if any(re.search(name, s, flags=re.IGNORECASE) for name in slot_names):
+                    # Watchlist slot 's' matches one of the slot names for this item
+                    slot = s
+        item = ""
+        for i in items_to_check:
+            if i.lower() in wqlist_item["name_enus"].lower():
+                # Watchlist item 'i' matches the name of this item
+                item = i
+        if item or slot:
+            slot = slot_names[0]  # slot_names[0] is the "official" name
+            yield (wqlist_item["name_enus"], item, slot)
 
 
 def _lookupQuest(quest_id, wqs_data):
-    # Takes a quest id and looks up quest in wqs_data. Returns quest name.
-    patt = '_[' + str(quest_id) + ']='
+    # Takes a quest id and looks up quest in wqs_data. Returns quest name and quest side since some wqs are for only one faction.
+    # TODO: Add a way to let the user ignore WQs that are specific to one faction
+    patt = "_[" + str(quest_id) + "]="
     quest_start = wqs_data.index(patt) + len(patt)
-    quest_end = wqs_data.index(';', quest_start)
+    quest_end = wqs_data.index(";", quest_start)
     quest_json = wqs_data[quest_start:quest_end]
 
     quest = json.loads(quest_json)
-    return quest['name_enus']
+    return quest["name_enus"], quest["_side"]
 
 
 def _formatTime(time):
```

Comment:
Added timezone logic.

This isn't a code comment, but I just wanna complain about timezones:
    * Daylight savings is dumb. 
    * Handling timezones in python is dumb.
    * Needing a 3rd party library to do daylight savings in python is dumb.
While I'm at it, it's dumb that we don't use 24 hour time more often in this country. How is `8:00 AM/PM` not more confusing than `8:00/20:00`?
Month/day/year is dumb too. It should be year/month/day.
I don't feel strongly enough about it to enforce my views on the output here contrary to common usage. I just wanted to rant.

Reply:
lol agreed. At least python is slightly better than JS, which 0-indexes its months, but 1-indexes its days...?
I realized at some point there would be a problem with time zones in the output, but it seemed like too much work to figure out what time zone the data was in to convert it. If you can figure it out, go for it!
Also, relevant: https://www.youtube.com/watch?v=-5wpm-gesOY

Reply:
Well JS is just dumb and weird in general so that's no surprise 😛:
https://www.destroyallsoftware.com/talks/wat

Also, relevant XKCDs:
https://xkcd.com/1883/
https://xkcd.com/2050/

Reply:
Wow.. those JS examples physically hurt to watch lol
And they made me think of another xkcd https://xkcd.com/1537/